dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/ingest/IngestServiceBean.java

```diff
@@ -850,21 +855,29 @@ public boolean ingestAsTabular(Long datafile_id) {
             } else {
                 tabDataIngest = ingestPlugin.read(inputStream, null);
             }
-        } catch (IOException ingestEx) {
+        } catch (IngestException ex) {
             dataFile.SetIngestProblem();
-            FileUtil.createIngestFailureReport(dataFile, ingestEx.getMessage());
-            dataFile = fileService.save(dataFile);
 
-            logger.warning("Ingest failure (IO Exception): " + ingestEx.getMessage() + ".");
+            if (ex.getBundleKey().isDefined() && ex.getBundleArguments().isEmpty()) {
+                FileUtil.createIngestFailureReport(dataFile, ex.getMessage(), ex.getBundleKey().get());
+            } else if (ex.getBundleKey().isDefined() && !ex.getBundleArguments().isEmpty()) {
+                FileUtil.createIngestFailureReport(dataFile, ex.getMessage(), ex.getBundleKey().get(), ex.getBundleArguments().toArray(new String[0]));
+            } else {
+                FileUtil.createIngestFailureReport(dataFile, ex.getMessage());
+            }
```

Comment:
Why not instead have a method `FileUtil#createIngestFailureReport(DataFile, IngestException)` and do all those iffy logic inside it?